projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-button.php

```diff
@@ -156,5 +161,135 @@ function sharing_process_requests() {
+		if (
+			! $global['show']
+			|| in_array( 'post', $global['show'], true )
+			|| in_array( 'page', $global['show'], true )
+		) {
+			return $hooked_block_types;
+		}
+	} elseif ( ( new Modules() )->is_active( 'sharedaddy' ) ) {
+		return $hooked_block_types;
+	}
+
+	// Only hook into page and single post templates.
+	if (
+		! $context instanceof \WP_Block_Template
+		|| ! property_exists( $context, 'slug' )
+		|| empty( $context->slug )
+		|| ! preg_match( '/^(page|single)/', $context->slug )
+	) {
+		return $hooked_block_types;
+	}
+
+	$content = $context->content ?? '';
+	// Check if the block is already in the template. If so, abort.
+	if ( false !== strpos( $content, 'wp:' . PARENT_BLOCK_NAME ) ) {
+		return $hooked_block_types;
+	}
+
+	// Add the block at the end of the post content.
+	if (
+		'after' === $relative_position
+		&& 'core/post-content' === $anchor_block_type
+	) {
+		$hooked_block_types[] = PARENT_BLOCK_NAME;
+	}
+
+	return $hooked_block_types;
+}
+add_filter( 'hooked_block_types', __NAMESPACE__ . '\add_block_to_single_posts_template', 10, 4 );
+
+/**
+ * Add default services to the block we add to the post content by default.
+ *
+ * @since $$next-version$$
+ *
+ * @param array                   $parsed_hooked_block The parsed block array for the given hooked block type.
+ * @param string                  $relative_position   The relative position of the hooked block.
+ * @param array                   $parsed_anchor_block The anchor block, in parsed block array format.
+ * @param WP_Block_Template|array $context             The block template, template part, or pattern that the anchor block.
+ *
+ * @return array
+ */
+function add_default_services_to_block( $parsed_hooked_block, $relative_position, $parsed_anchor_block, $context ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+	// Is the hooked block adjacent to the anchor block?
+	if ( 'after' !== $relative_position ) {
+		return $parsed_hooked_block;
+	}
+
+	// Use the icon style by default.
+	$parsed_hooked_block['attrs']['styleType'] = 'icon';
+
+	// Does the anchor block have a layout attribute?
+	if ( isset( $anchor_block['attrs']['layout'] ) ) {
+		// Copy the anchor block's layout attribute to the hooked block.
+		$parsed_hooked_block['attrs']['layout'] = $parsed_anchor_block['attrs']['layout'];
+	}
+
+	// Add default services (inner blocks) to the block.
+	$parsed_hooked_block['innerBlocks'] = array(
+		array(
+			'blockName' => INNER_BLOCK_NAME,
+			'attrs'     => array(
+				'service' => 'facebook',
+				'label'   => esc_html__( 'Facebook', 'jetpack' ),
+			),
+		),
+		array(
+			'blockName' => INNER_BLOCK_NAME,
+			'attrs'     => array(
+				'service' => 'x',
+				'label'   => esc_html__( 'X', 'jetpack' ),
+			),
+		),
+		array(
+			'blockName' => INNER_BLOCK_NAME,
+			'attrs'     => array(
+				'service' => 'mastodon',
+				'label'   => esc_html__( 'Mastodon', 'jetpack' ),
+			),
+		),
+	);
```

Comment:
The "parsed block" format is a bit weird; if you have `innerBlocks`, you also need to set `innerContent`. 

The latter is an array whose items are either HTML strings, or `null`, which tells WP/GB to substitute the `null` for the respective next block in the `innerBlocks` array.

That might sound a bit abstract; here's an example from https://github.com/WordPress/wordpress-develop/pull/5835 to illustrate:

```
// $hooked_block is something like array( 'blockName' => 'ockham/like-button', 'attrs' => array() );

// Wrap the Like Button block in a Group block.
return array(
	'blockName' => 'core/group',
	'attrs'     => $attrs,
	'innerBlocks' => array( $hooked_block ),
	'innerContent' => array(
		'<div class="wp-block-group">',
		null,
		'</div>'
	),
);
```

So at the very least, you'll need to set `innerContent => array( null, null, null )` (as you're trying to insert 3 inner blocks).

But you might need more than that: If you insert a `jetpack/sharing-buttons` manually in the editor and then view it in the code editor, does it include any HTML? If so, you'll have to include those HTML strings in `innerContent`.

Reply:
Thanks for the quick feedback! That was helpful, things are starting to look better with c0445f06fab289408d7b0cebe873bc4a2a7dd4ba :)